input/workerpool.go

```diff
@@ -0,0 +1,35 @@
+package input
+
+import (
+	"sync"
+	"time"
+)
+
+type workerPool struct {
+	wg       sync.WaitGroup
+	shutdown chan struct{}
+}
+
+// registers this instance in the list of stoppables
+// also initializes the shutdown channel
+func (w *workerPool) startStoppable() {
+	stoppables = append(stoppables, w)
+	w.shutdown = make(chan struct{})
+}
```

Comment:
this is hard to follow. instance means "a pool of workers", which adds itself to a list which is defined in a different file?
and "startstoppable" is not a verb.

Reply:
this is too abstract.
I think the caller (e.g. the main command) should maintain a list of plugins. and just add them to the list when it creates them.
and the input plugins can have their own stop methods and waitgroups if they need them, instead of embedding this vague separate type. and the instances in the input package should also not have their own determination of when they timeout, that can be handled by the caller, as pointed out in another comment.

this is basically how MT does it, and i think it's easier to follow

Reply:
just as an explanation of the name:
`Stoppable` is the type of the object that have a `.stop()` method, this method starts the `Stoppable`, so `startStoppable()`.